subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/ConfigurationDependencyMetadataWrapper.java

```diff
@@ -61,7 +63,11 @@ public ConfigurationDependencyMetadataWrapper(ConfigurationMetadata configuratio
 
     @Override
     public List<ExcludeMetadata> getExcludes() {
-        return delegate.getConfigurationExcludes(configuration.getHierarchy());
+        if (cachedExcludes != null) {
+            return cachedExcludes;
+        }
+        cachedExcludes = delegate.getConfigurationExcludes(configuration.getHierarchy());
+        return cachedExcludes;
```

Comment:
This makes me wonder if we shouldn't do more processing when we parse the original `ivy.xml` into the internal representation, so that we cache them in a more efficient form.

Reply:
Yeah, if we could avoid computing the configuration excludes, that would be great. However, this may be a flaw of this specific test, which contains a large number of Ivy dependencies. It may not represent what we see most in the wild.

Reply:
+1 for handling this closer to the source instead of caching it here.